### PR TITLE
Fixed NPE in Forwarder, improved test stability, activated code in DeferCloseHttpClient

### PR DESCRIPTION
**Pull request by dominik-cnx**
As per description there are three issues addressed in this PR

### CODE REVIEW

File: gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java [hunk 1]
```diff
@@ -40,65 +42,65 @@ public void request(HttpMethod method, int port, String host, String requestURI,
...
-//            Handler<Throwable> originalExceptionHandler = getExceptionHandler(upstreamRsp);
-//            upstreamRsp.exceptionHandler(event -> {
-//                logger.debug("upstreamRsp.exceptionHandler({})", event.toString());
-//                if (needToDecrementCounter.getAndSet(false)) {
-//                    onEndOfRequestResponseCycle();
-//                }
-//                callHandlerIfExists(originalExceptionHandler, event);
-//            });
-//        }
+        delegate.request(method, port, host, requestURI).onComplete(asyncRequestResult -> {
```
Comment (dominik-cnx) [inline gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java:45]: @hiddenalpha do we need to manually test this in order to know if it still works?

File: gateleen-cache/src/test/java/org/swisspush/gateleen/cache/storage/RedisCacheStorageTest.java [hunk 2]
```diff
@@ -154,6 +154,9 @@ public void testCacheEntriesCount(TestContext context) {
 
         // prepare
         jedis.sadd(CACHED_REQUESTS, "cache_item_1", "cache_item_2", "cache_item_3");
+        jedis.set(CACHE_PREFIX + "cache_item_1", jsonObjectStr("payload_1"));
```
Comment (dominik-cnx) [inline gateleen-cache/src/test/java/org/swisspush/gateleen/cache/storage/RedisCacheStorageTest.java:157]: @mcweba kindly review if this makes sense. I think the sequence of the test execution is random in here.

File: gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java (hunk 1 above)
Comment (hiddenalpha) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java:45]: If I remind correctly, I wrote the [RuleUpdateTest](https://github.com/swisspush/gateleen/blob/v1.3.0/gateleen-test/src/test/java/org/swisspush/gateleen/routing/RuleUpdateTest.java#L26) for that purpose.

Comment (Swisspusher) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java:45]: The RuleUpdateTest can pass with out this, I mean already passes before enabled this part of code, so start to confuse.

Comment (ZhengXinCN) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java:45]: The RuleUpdateTest can pass with out this, I mean already passes before enabled this part of code, so start to confuse.

Comment (dominik-cnx) [reply gateleen-routing/src/main/java/org/swisspush/gateleen/routing/DeferCloseHttpClient.java:45]: I think @ZhengXinCN is right, it passed in v1.3.0 here (the code is commented in this version)

https://api.travis-ci.com/v3/job/561702838/log.txt

[INFO] Running org.swisspush.gateleen.routing.RuleUpdateTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 63.751 s - in org.swisspush.gateleen.routing.RuleUpdateTest

maybe the behavior has somehow changed now, but not sure what exactly. I think we should further look into this.

**Pull request merged**